src/watchmaker/__init__.py

```diff
@@ -51,42 +47,73 @@ def __init__(self, arguments):
                 Should an s3 bucket be used for the installation files.
             config_path (str):
                 Path to YAML configuration file.
-            logger (bool):
-                Enables self.logger to a file.
-            log_path (str):
-                Path to logfile for stream self.logger.
+            log_dir (str) or log_file (str):
+                Path to log directory or file for stream logging.
         """
         self.kwargs = {}
         self.noreboot = arguments.noreboot
         self.s3 = arguments.sourceiss3bucket
         self.system = platform.system()
         self.config_path = arguments.config
         self.default_config = os.path.join(static.__path__[0], 'config.yaml')
-        self.log_path = arguments.log_path
         self.saltstates = arguments.saltstates
         self.config = None
         self.system_params = None
         self.system_drive = None
         self.execution_scripts = None
-        logging.basicConfig()
-        self.logger = logging.getLogger()
 
-        if arguments.logger and os.path.exists(arguments.log_path):
+        self._prepare_logger(arguments.log_dir, arguments.log_file)
+        logging.info('Parameters:  {0}'.format(self.kwargs))
+        logging.info('System Type: {0}'.format(self.system))
+
+    def _prepare_logger(self, log_dir, log_file):
+        """
+        Prepares the logger for handling messages to a file and/or to stdout.
+        Args:
+            log_dir (str):
+                Path of a directory. If this object is not None, then this
+                directory will be used to store a log file.
+            log_file (str):
+                Path to a file. If this object is not None, then this path
+                to a file will be used as the log file.
+        """
+        log_filename = None
+        if log_dir and os.path.exists(log_dir):
+            if os.path.isfile(log_dir):
+                log_msg = '{0} is a file and not a directory.'.format(log_dir)
+                log_type = 'error'
+            else:
+                log_filename = os.path.join(
+                    log_dir,
+                    'watchmaker-{0}.log'.format(str(datetime.date.today()))
+                )
+                log_msg = 'Start time: {0}'.format(datetime.datetime.now())
+                log_type = 'info'
+        elif log_file:
+            if os.path.isdir(log_file):
+                log_msg = '{0} is a directory and not a file.'.format(log_file)
+                log_type = 'error'
+            else:
+                log_filename = log_file
+                log_msg = 'Start time: {0}'.format(datetime.datetime.now())
+                log_type = 'info'
+        elif not log_dir and not log_file:
+            log_msg = 'Stream logger is not enabled!'
```

Comment:
I don't think users are going to know what a stream logger is. (I still don't....) What does this really mean, from the user's perspective?